components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -132,11 +132,31 @@ public void removeClaimDialect(ClaimDialect claimDialect, String tenantDomain) t
         // Add listener
 
         this.claimDialectDAO.removeClaimDialect(claimDialect, tenantId);
+        postRemoveClaimDialect(claimDialect.getClaimDialectURI(), tenantId);
```

Comment:
This needs to fixed at the caching level. Need to check why the cache is not cleared properly

Reply:
Add method to remove external claim cache when deleting the claim dialect in "CacheBackedExternalClaimDAO" class. https://github.com/wso2/carbon-identity-framework/pull/2778/commits/e1c21e4270cadf8606625640c51d27da21416696